src/mochiweb_http.erl

```diff
@@ -168,8 +174,14 @@ collect_headers(Socket, Request, Body, Collected, Trunc, HeaderCount) ->
         {ssl_closed, _} ->
             mochiweb_socket:close(Socket),
             exit(normal);
-        _Other ->
-            handle_invalid_request(Socket, Request, [])
+        {ProtocolErr, _Socket, _Reason} when ProtocolErr =:= tcp_error orelse ProtocolErr =:= ssl_error ->
```

Comment:
small risk our spurious message is a tcp_error about another socket, but probably too irritating to dig out the SSL vs TCP port delivering it.  Closing the socket is a strong indicator, and nothing has been written to the output yet - still collecting headers.